pkg/profiles/profiles.go

```diff
@@ -306,6 +315,90 @@ func ToRequestMatch(reqMatch *sp.RequestMatch) (*pb.RequestMatch, error) {
 	}, nil
 }
 
+// ValidateSP takes a v1alpha1.ServiceProfile object, marshals to bytes, and
+// wraps `Validate`.
+func ValidateSP(serviceProfile sp.ServiceProfile) error {
+	data, err := yaml.Marshal(serviceProfile)
+	if err != nil {
+		return fmt.Errorf("failed to marshall ServiceProfile [%+v]: %s", serviceProfile, err)
+	}
+	return Validate(data)
+}
+
+// Validate validates the structure of a ServiceProfile. This code is a superset
+// of the validation provided by the `openAPIV3Schema`, defined in the
+// ServiceProfile CRD.
+// openAPIV3Schema validates:
+// - types of non-recursive fields
+// - presence of required fields
+// This function validates:
+// - types of all fields
+// - presence of required fields
+// - presence of unknown fields
+// - recursive fields
+func Validate(data []byte) error {
+	var serviceProfile sp.ServiceProfile
+	err := yaml.UnmarshalStrict([]byte(data), &serviceProfile)
+	if err != nil {
+		return fmt.Errorf("failed to validate ServiceProfile: %s", err)
```

Comment:
It would be helpful to have the name of the service profile in this error message.  It's not available here since the unmarshal failed, but it is available in the health check this gets called from.  We could have the health wrap the error message with the profile name and also remove the profile name from the other errors in this method.

Reply:
I opted to modify the healthcheck to include the ServiceProfile name, but leave the underlying errors as-is. Although this means error messages may print the name twice, I am assuming when we start calling `profiles.Validate` from a `ValidatingAdmissionWebhook`, we will want the name included in the error message.